bokeh/core/property/instance.py

```diff
@@ -170,3 +172,4 @@ def _sphinx_type_link(obj: Instance[Any]) -> str:
     model = obj.instance_type
```

Comment:
I think this would solve the problem:
```
    if isinstance(obj._instance_type, types.FunctionType):
        return f"{property_link(obj)}"

    if isinstance(obj._instance_type, str):
        return f"{property_link(obj)}({obj._instance_type})"

    model = obj._instance_type
```